# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/thebes/gfxPlatform.h
# Commit: 0facab7b9a1d
# Full Hash: 0facab7b9a1dad599c9f5049cdf104926eace696
# Author: Kelsey Gilbert <kelsey.gilbert@mozilla.com>
# Date: 2022-12-21 21:21:23
# Regressor Bug: 1799258
# File Overlap Count: 1
# Description:
#   Bug 1799258 - Do color-management on Windows+DComp via IDCompositionFilterEffects. r=sotaro
#   
#   + Add gfx.color_management.rec709_gamma_as_srgb:true. :'(
#   
#   In particular, rec709(16/255) -> srgb(31/255). Even though it's
# ==============================================================================

diff -r 4f9ee3537468 -r 0facab7b9a1d gfx/thebes/gfxPlatform.h
--- a/gfx/thebes/gfxPlatform.h	Tue Dec 20 22:47:52 2022 +0000
+++ b/gfx/thebes/gfxPlatform.h	Tue Dec 20 22:47:53 2022 +0000
@@ -751,7 +751,9 @@
    * Returns a buffer containing the CMS output profile data. The way this
    * is obtained is platform-specific.
    */
-  virtual nsTArray<uint8_t> GetPlatformCMSOutputProfileData();
+  virtual nsTArray<uint8_t> GetPlatformCMSOutputProfileData() {
+    return GetPrefCMSOutputProfileData();
+  }
 
   /**
    * Return information on how child processes should initialize graphics
@@ -863,13 +865,15 @@
   virtual void ImportContentDeviceData(
       const mozilla::gfx::ContentDeviceData& aData);
 
+ public:
   /**
    * Returns the contents of the file pointed to by the
    * gfx.color_management.display_profile pref, if set.
    * Returns an empty array if not set, or if an error occurs
    */
-  nsTArray<uint8_t> GetPrefCMSOutputProfileData();
+  static nsTArray<uint8_t> GetPrefCMSOutputProfileData();
 
+ protected:
   /**
    * If inside a child process and currently being initialized by the
    * SetXPCOMProcessAttributes message, this can be used by subclasses to